### ISSUE TITLE
Pass default np float array to c++ should be float32.

### ISSUE THREAD

**Issue by lionlai1989** [cmla/s2p#85]
When calling funtion `lib.remove_isolated_3d_points(np.ascontiguousarray(xyz), w, h, r, p, n, q)`, it shows an error
`ctypes.ArgumentError: argument 1: <class 'TypeError'>: array must have data type float32`

I believe the issue here is that `remove_isolated_3d_points` expects a float32 array but the default np array is float64.
```
    lib.remove_isolated_3d_points.argtypes = (
        ndpointer(dtype=c_float, shape=(h, w, 3)),
        c_int, c_int, c_float, c_int, c_int, c_int)
```
So, convert the float64 to float32 solves the issue.
`lib.remove_isolated_3d_points(np.float32(np.ascontiguousarray(xyz)), w, h, r, p, n, q)`
`lib.count_3d_neighbors(out, np.float32(np.ascontiguousarray(xyz)), w, h, r, p)`

@gfacciol @carlodef Please take a look at this. Thanks a lot.

**Comment by gfacciol** [comment]
Thanks @lionlai1989 indeed this problem is related to the data type. We introduced it with #82.  
The change to float64 must be propagated to the library because the point cloud xyz must be stored in float64.

**Comment by lionlai1989** [comment]
@gfacciol But the c function `remove_isolated_3d_points`still expects a float32 array, so I think it should be 
```
    lib.remove_isolated_3d_points.argtypes = (
        ndpointer(dtype=c_double, shape=(h, w, 3)),
        c_int, c_int, c_float, c_int, c_int, c_int)
```
`dtype=c_float` should be `dtype=c_double`?

**Comment by lionlai1989** [comment]
According to https://docs.python.org/3/library/ctypes.html, the c_float is 32 bits instead of 64 bits.

**Comment by gfacciol** [comment]
Indeed,  the C functions in disp_to_h.c  must be changed to be double-typed  
https://github.com/cmla/s2p/blob/7121ee3f299e163e76f6c92c8c8fc163b02835f2/c/disp_to_h.c#L169

remove_isolated_3d_points
count_3d_neighbors
squared_distance_between_3d_points
and the interfaces

**Comment by carlodef** [comment]
Thank you @lionlai1989! This should be fixed by #86
